### PR TITLE
Details popup window

### PR DESCRIPTION
**Pull request by dvcorreia1984**
In this pull request the following was done: 

1. The project section was refactored using an Javascript array to produce content dynamically. 
2. An popup modal was added responding differently in the mobile and desktop version of the app. 
3. ES Lint errors were also corrected.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -78,97 +78,15 @@ <h1 class="heading1">
...
-                              <li><div class="button2"><button2>Ruby on Rails</button2><div><li>
-                              <li><div class="button2"><button2>CSS</button2><div><li>  
-                              <li><div class="button2"><button2>Javascript</button2><div><li>
-                              <li><div class="button2"><button2>HTML</button2><div><li>
-                            </ul>
-                            <button class="button">See Project</button>
-                        </div>
-                      </div>
-                    </div>
+      <div id="projects-container" class=grid-container>
```
Comment (Kaiserabbas) [inline index.html:81]: Kindly consider creating new elements through javascript 
 as HTML for the project section needs to be loaded through Javascript file.

Comment (dvcorreia1984) [reply index.html:81]: Ok Kaiser, thanks.

Comment (dvcorreia1984) [reply index.html:81]: The reviewer asked us to do morning session notes in the Issues section. Please take note of this for future reference.

File: styles.css [hunk 2]
```diff
@@ -922,3 +956,220 @@ div#sidenav-line {
...
+  justify-content: flex-start;
+  gap: 0.4rem;
+  padding: 1.3rem;
+  min-height: 250px;
+  position: absolute;
+  background-color: white;
+  border: 1px solid #ddd;
+  border-radius: 15px;
+  z-index: 2;
+}
```
Comment (AbelG101) [inline styles.css]: - [ ] Kindly make sure the design of the popup modal follows the given Figma design as I noticed there are some issues in your design as outlined below 👇
- Your design:
  image
firefox_AQPzMJocDm
- Figma design:
  image
- [ ] I've mentioned some of the issues in the screenshot but I highly advise you to pay attention to the overall design (like paragraph color...) as a frontend developer you have to master such skills so you have a better chance of impressing future clients.

File: styles.css
Comment (AbelG101) [inline styles.css:1]: - [ ] For better UX I'd highly advise you center the popup modal as it's currently appearing at the bottom of the screen which can confuse users.
- Here's how it looks:
  firefox_P7Yi5iFkL4
- This applies to the mobile version as well. ✔
- [ ] I also noticed a bug in the popup modal, when clicking the "see project" button multiple times the popup modal is being rendered multiple times as well, see the below video 👇
  firefox_bGpm2a8Srz

Comment (AbelG101) [inline styles.css:1]: - [ ] I noticed the project cards in the mobile version don't close when the close button is clicked so I'd highly advise you to fix it.
- Here's how it looks:
  firefox_1wrjAEHRgf

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @dvcorreia1984  👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Correct use of GitHub Flow ✔
-  Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: styles.css (hunk 2 above)
Comment (dvcorreia1984) [reply styles.css]: I made these changes.

File: styles.css
Comment (dvcorreia1984) [reply styles.css:1]: I fixed the bug, but still need to work out how to centre the modal.

Comment (dvcorreia1984) [reply styles.css:1]: I fixed the issue with the button.

Comment (dvcorreia1984) [reply styles.css:1]: I centered the modal position...

**Review by topeogunleye — APPROVED** [review]
### Hi @dvcorreia1984, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Fixed Popup Modals ✔️ 
 
## [Optional] suggestions:
- For future pull requests, please notice that your branch name `details-popup-window` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-details-popup-window`. This makes it easier to track the changes and the purpose of each branch. 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**